include/networkit/graph/Graph.hpp

```diff
@@ -683,6 +683,14 @@ class Graph final {
     /** Default copy assignment operator */
     Graph &operator=(const Graph &other) = default;
 
+	/**
+	 * Reserves memory in the node's edge containers.
+	 *
+	 * @param u the node memory should be reserved for 
+	 * @param size the amount of memory to reserve
+	 * */
+	void preallocate(node u, size_t size);
```

Comment:
I would like to use this function in a parallel context. Can we please add the guarantee to the documentation that it is thread-safe to use as long as two concurrent invocations to not access the same node u. The current implementation allows this without modification.

Reply:
Agreed.